### DISCUSSION TITLE
Energy dashboard integration

### DISCUSSION THREAD

**Discussion by konagy** [djansen1987/SAJeSolar#35] [Q&A]
Hi all,

Is there any way that I can integrate the SAJ inverter to the HASS Energy dashboard. I can only find `totalElectricity` entity available and I'm wondering if more (maybe all) entities are suitable for integration into the energy dashboard.

Thx,
konagy

**Comment by imageone10** [comment]
I am very interested in this.
I have tried to integrate the values of my SAJ H1 in Home Assistant through the energy panel and through the "tesla style solar powercard" without success because it only recognizes the total accumulated values as a sensor. It's nice to be able to read cumulative daily values on my dashboard, but I think it's more useful to have an instant (or near-instant) reading of solar production, consumption, and storage values.

**Comment by djansen1987** [comment]
Hi It should be able to add your SAJ inverter in the Energy dashboard. The energy dashboard only accepts A total sensor with an increasing value. Could you please explain what you are missing or show screenshots of what is not working for you ?

**Comment by imageone10** [comment]
The point is that it doesn't make much sense to have a flow diagram that shows how energy flows in one direction and another (from photovoltaic panels to the grid, from the grid to the house...) and that diagram shows only the accumulated daily total values.
The interesting thing about a diagram of this type is that it shows the instantaneous values, in order to be able to see at a glance the amount of energy produced by the photovoltaic panels, the amount that the house is consuming and the amount that is being used to charge batteries or feed from them to the home, that way you can modify the consumption of your home to adapt them to photovoltaic generation.
In my case, I have had no problems adding the SAJ e-solar integration to my Home Assistant. In that aspect it works correctly and I thank you for the work you have done to achieve it.
I think it would be great to be able to use sensors like `sensor.esolar_nowpower` or `sensor.esolar_batterypower` in the Home Assistan energy dashboard, but from what I see you can only use total energy sensors like `sensor.esolar_totalelectricity`
There is a card that fulfills this task (https://github.com/reptilex/tesla-style-solar-power-card)
The problem that I find is that when defining the sensors and creating custom sensors I do not have enough knowledge to do it. I don't know how to combine, for example, sensor.esolar_batterypower with sensor.esolar_batterydirection to indicate that the watts it reads in the first are charging or discharging (positive or negative reading) so that it represents the flow in one direction or another. I also have no way of indicating whether `sensor.esolar_gridpower` is pouring energy into the grid or taking from it.
It would be great if you could help us to execute this task or modify the sensors to be able to provide the necessary readings on this card.
I think that if each one of the sensors that measures energy or power could deliver the value in positive or negative, it would be possible to integrate some of those energy cards.
Sensors like: (batteryPower, gridPower, totalLoadPower)
